frontend/src/hooks/useClientFetch.ts

```diff
@@ -18,25 +18,25 @@ export const useClientFetch = <T>(
   const { refreshIfExpired, refreshUser } = useUser();
   const router = useRouter();
 
-  const fetchWithAuthCheck = async (
-    url: string,
-    options: RequestInit = {},
-  ): Promise<Response> => {
-    const expired = await refreshIfExpired();
-    if (expired && authGatedRequest) {
-      router.refresh();
-      throw new Error("local token expired, logging out");
-    }
-    const response = await fetch(url, options);
-    if (response.status === 401) {
-      await refreshUser();
-      if (authGatedRequest) {
+  const fetchWithAuthCheck = useCallback(
+    async (url: string, options: RequestInit = {}): Promise<Response> => {
+      const expired = await refreshIfExpired();
+      if (expired && authGatedRequest) {
         router.refresh();
+        throw new Error("local token expired, logging out");
+      }
+      const response = await fetch(url, options);
+      if (response.status === 401) {
+        await refreshUser();
+        if (authGatedRequest) {
+          router.refresh();
+        }
+        return response;
       }
       return response;
-    }
-    return response;
-  };
+    },
+    [authGatedRequest, refreshIfExpired, refreshUser, router],
```

Comment:
why did these need to get updated?

Reply:
@doug-s-nava to address warnings that were being thrown by the linter

Reply:
in order to limit the scope on this, if you could remove the useCallback for now that'd be best